### PR TITLE
feat: add kmskeyring ref

### PR DESCRIPTION
**Pull request by acpana**
Realized I am not going to use the `KmsKeyRef` for `DataformRepository` as it's not part of their proto definition just yet. So parking my work here for now.

### CODE REVIEW

File: apis/refs/v1beta1/helper.go [hunk 1]
```diff
@@ -30,3 +30,14 @@ func GetResourceID(u *unstructured.Unstructured) (string, error) {
 	}
 	return resourceID, nil
 }
+
+func GetLocation(u *unstructured.Unstructured) (string, error) {
```
Comment (gemmahou) [inline apis/refs/v1beta1/helper.go:34]: We don't want to make this a helper function, see: https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/2337#discussion_r1696037535

File: apis/refs/v1beta1/kmskeyref.go
Comment (gemmahou) [inline apis/refs/v1beta1/kmskeyref.go:1]: I wanted to group refs and their resolver functions in one file by service... i.e. my compute refs: https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/apis/refs/v1beta1/computerefs.go

But I think your approach is clear, I'm open to suggestions.

File: apis/refs/v1beta1/kmskeyref.go [hunk 2]
```diff
@@ -14,11 +14,97 @@
...
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
 type KMSCryptoKeyRef struct {
-	/* The KMSCryptoKeyRef selfLink, when not managed by KCC. */
+	//  If provided must be in the format `projects/[kms_project_id]/locations/[region]/keyRings/[key_ring_id]/cryptoKeys/[key]`.
```
Comment (gemmahou) [inline apis/refs/v1beta1/kmskeyref.go:30]: I've been thinking about this, I feel checks on External is unnecessary. KCC does not validate External format for now, and for example ComputeAddressRef, its External field accepts ID, full URL, or an address. I do not think KCC has a fixed pattern for External field, and should let GCP to fail the request if the value if invalid. I'll have a PR to discuss this, before that let's process with the current approach.

File: apis/refs/v1beta1/kmskeyref.go [hunk 3]
```diff
@@ -14,11 +14,97 @@
...
 	External string `json:"external,omitempty"`
-	/* The `name` field of a `KMSCryptoKeyRef` resource. */
+	// The `name` field of a `KMSCryptoKey` resource.
```
Comment (gemmahou) [inline apis/refs/v1beta1/kmskeyref.go:32]: Good catch!!

File: apis/refs/v1beta1/helper.go (hunk 1 above)
Comment (acpana) [reply apis/refs/v1beta1/helper.go:34]: But it's going to be used a fair amount I think for any ref that has `locations` in its fully qualified name

**Pull request merged**